dask_ml/model_selection/_incremental.py

```diff
@@ -347,3 +383,384 @@ def fit(
+
+
+class RandomIncrementalSearch(BaseIncrementalSearch):
+    def __init__(
+        self,
+        estimator,
+        param_distribution,
+        n_iter=10,
+        max_iter=100,
+        patience=10,
+        tol=0.001,
+        test_size=0.15,
+        random_state=None,
+        scoring=None,
+        iid=True,
+        refit=True,
+        error_score="raise",
+        return_train_score=_RETURN_TRAIN_SCORE_DEFAULT,
+        scheduler=None,
+        n_jobs=-1,
+        cache_cv=True,
+    ):
+        self.max_iter = max_iter
+        self.n_iter = n_iter
+        self.patience = patience
+        self.tol = tol
+        super(RandomIncrementalSearch, self).__init__(
+            estimator,
+            param_distribution,
+            test_size,
+            random_state,
+            scoring,
+            iid,
+            refit,
+            error_score,
+            return_train_score,
+            scheduler,
+            n_jobs,
+            cache_cv,
+        )
+
+    def _get_params(self):
+        return ParameterSampler(self.parameters, self.n_iter)
+
+    def _additional_calls(self, info):
+        out = {}
+        max_iter = self.max_iter
+        patience = self.patience
+        tol = self.tol
+
+        for ident, records in info.items():
+            if max_iter is not None and len(records) > max_iter:
+                out[ident] = 0
+
+            elif len(records) > patience:
+                old = records[-patience]["score"]
+                if all(d["score"] < old + tol for d in records[-patience:]):
+                    out[ident] = 0
+                else:
+                    out[ident] = 1
+
+            else:
+                out[ident] = 1
+        return out
+
+
+class SuccessiveReductionSearch(BaseIncrementalSearch):
+    def __init__(
+        self,
+        estimator,
+        param_distribution,
+        n_iter=10,
+        max_iter=100,
+        test_size=0.15,
+        decay_rate=1.0,
+        random_state=None,
+        scoring=None,
+        iid=True,
+        refit=True,
+        error_score="raise",
+        return_train_score=_RETURN_TRAIN_SCORE_DEFAULT,
+        scheduler=None,
+        n_jobs=-1,
+        cache_cv=True,
+    ):
+        self.max_iter = max_iter
+        self.n_iter = n_iter
+        self.decay_rate = decay_rate
+        super(SuccessiveReductionSearch, self).__init__(
+            estimator,
+            param_distribution,
+            test_size,
+            random_state,
+            scoring,
+            iid,
+            refit,
+            error_score,
+            return_train_score,
+            scheduler,
+            n_jobs,
+            cache_cv,
+        )
+
+    def _get_params(self):
+        return ParameterSampler(self.parameters, self.n_iter)
+
+    def _additional_calls(self, info):
+        def inverse(time):
+            """ Decrease target number of models inversely with time """
+            return int(self.n_iter / (1 + time) ** self.decay_rate)
```

Comment:
Should this be `self.max_iter`? `self.n_iter` is used for the initial sampling. (we probably need more distinct names for these)

Reply:
Using `self.n_iter` is correct here, this is a decaying exponential starting at the number of initial parameters.